pkg/ottl/compare.go

```diff
@@ -136,6 +136,17 @@ func (p *Parser[K]) compareFloat64(a float64, b any, op compareOp) bool {
 	}
 }
 
+func (p *Parser[K]) compareDuration(a time.Duration, b any, op compareOp) bool {
```

Comment:
What's the benefit of have `b` as `any` type? I see that's being done with the other comparisons as well, so I'm curious why we choose `any` here instead of making it `time.Duration`?

Reply:
The comparison logic flow first checks the type of the left side of the comparison, decides which comparison function to use, and then checks the type of B. This helps us combine some logic prevent nested switch statements.

Reply:
We use the item for both `time.Time` and `time.Duration` so it cannot only conform to one of those types.

Reply:
That's makes a lot of sense, thank you!